Jenkinsfile

```diff
@@ -39,68 +39,67 @@ String getJenkinsImageTag() {
   return sh(returnStdout: true, script: "grep ImageTag values.yaml | awk '{print \$2}'")
 }
 
+String getTargetNamespace() {
+  return BRANCH_NAME == 'master' ? 'platform' : 'platform-staging'
+}
+
 pipeline {
   agent {
     label 'jenkins-jx-base'
   }
   environment {
+    NAMESPACE = getTargetNamespace()
     SERVICE_ACCOUNT = 'jenkins'
   }
   stages {
-    stage('Upgrade Jenkins X platform') {
+    stage('Upgrade Jenkins X') {
       steps {
-        setGitHubBuildStatus('upgrade', 'Upgrade Jenkins X platform', 'PENDING')
+        setGitHubBuildStatus('upgrade', 'Upgrade Jenkins X', 'PENDING')
         container('jx-base') {
-          echo "Upgrade Jenkins X platform using jenkins image tag ${getJenkinsImageTag()}"
+          echo "Upgrade Jenkins X in the ${NAMESPACE} namespace using jenkins image tag ${getJenkinsImageTag()}"
           script {
             // get the existing docker registry auth
-            def dockerRegistryConfig = sh(script: 'kubectl get secret jenkins-docker-cfg -o go-template=\$\'{{index .data "config.json"}}\' | base64 --decode | tr -d \'\\n\'', returnStdout: true).trim();
+            def dockerRegistryConfig = sh(script: 'kubectl get secret jenkins-docker-cfg -n ${NAMESPACE} -o go-template=\$\'{{index .data "config.json"}}\' | base64 --decode | tr -d \'\\n\'', returnStdout: true).trim();
             // get the existing nexus password
-            def nexusPassword = sh(script: 'kubectl get secret -o=jsonpath=\'{.data.password}\' nexus | base64 --decode', returnStdout: true)
-            // upgrade Jenkins platform
-            withEnv(["DOCKER_REGISTRY_CONFIG=${dockerRegistryConfig}", "NEXUS_PASSWORD=${nexusPassword}"]) {
+            def nexusPassword = sh(script: 'kubectl get secret nexus -n ${NAMESPACE} -o=jsonpath=\'{.data.password}\' | base64 --decode', returnStdout: true)
+            // upgrade Jenkins
+            withEnv([
+              "INTERNAL_DOCKER_REGISTRY=${DOCKER_REGISTRY}",
+              "DOCKER_REGISTRY_CONFIG=${dockerRegistryConfig}",
+              "NEXUS_PASSWORD=${nexusPassword}",
+            ]) {
               sh """
                 # initialize Helm without installing Tiller
                 helm init --client-only --service-account ${SERVICE_ACCOUNT}
 
                 # add local chart repository
                 helm repo add jenkins-x http://chartmuseum.jenkins-x.io
 
-                # replace env vars in values.yaml: DOCKER_REGISTRY, DOCKER_REGISTRY_CONFIG
-                envsubst < values.yaml > myvalues.yaml
+                # replace env vars in values.yaml
+                envsubst '\${NAMESPACE} \${INTERNAL_DOCKER_REGISTRY} \${DOCKER_REGISTRY_CONFIG} \${NEXUS_PASSWORD}' < values.yaml > myvalues.yaml
```

Comment:
Why do we need them as they are already present in the env it should be available for envsubst no?

Reply:
I'm specifying them explicitly to *not* replace `DOCKER_REGISTRY` which needs to reamain unreplaced to be relative to the upgraded namespace: `platform-staging` (PR) or `platform` (master). I'll add a comment.